src/libraries/System.Net.Primitives/src/System/Net/NetworkException.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Runtime.Serialization;
+
+namespace System.Net
+{
+    /// <summary>Provides socket exceptions to the application.</summary>
+    [Serializable]
+    public class NetworkException : IOException
+    {
+        /// <summary>Creates a new instance of the <see cref='System.Net.NetworkException'/> class with the specified error code.</summary>
+        protected NetworkException(NetworkError error, Exception? innerException = null, string? message = null)
+            : base(message ?? GetExceptionMessage(error), innerException)
+        {
+            NetworkError = error;
+        }
+
+        protected NetworkException(SerializationInfo serializationInfo, StreamingContext streamingContext)
+            : base(serializationInfo, streamingContext)
+        {
+        }
+
+        /// <summary>Returns the specific kind of error.</summary>
+        public NetworkError NetworkError { get; }
```

Comment:
The serialization ctor doesn't set this, and it's missing a GetObjectData override to store this. Is that on purpose?

Reply:
Nope, will add.